From: Jon Cooper <jcooper@solarflare.com>
Date: Tue, 21 Oct 2014 14:50:29 +0100
Subject: [PATCH v2 243/244] sfc: remove incorrect EFX_BUG_ON_PARANOID check
Patch-mainline: v3.18-rc3
Git-commit: 8fc963515e893867330dec87464e9edc5204c024
References: bsc#909618 FATE#317521

write_count and insert_count can wrap around, making > check invalid.

Fixes: 70b33fb0ddec827cbbd14cdc664fc27b2ef4a6b6 ("sfc: add support for
 skb->xmit_more").

Signed-off-by: Edward Cree <ecree@solarflare.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/sfc/tx.c |    4 ----
 1 file changed, 4 deletions(-)

--- a/drivers/net/ethernet/sfc/tx.c
+++ b/drivers/net/ethernet/sfc/tx.c
@@ -347,8 +347,6 @@ netdev_tx_t efx_enqueue_skb(struct efx_t
 	unsigned short dma_flags;
 	int i = 0;
 
-	EFX_BUG_ON_PARANOID(tx_queue->write_count != tx_queue->insert_count);
-
 	if (skb_shinfo(skb)->gso_size)
 		return efx_enqueue_skb_tso(tx_queue, skb);
 
@@ -1248,8 +1246,6 @@ static int efx_enqueue_skb_tso(struct ef
 	/* Find the packet protocol and sanity-check it */
 	state.protocol = efx_tso_check_protocol(skb);
 
-	EFX_BUG_ON_PARANOID(tx_queue->write_count != tx_queue->insert_count);
-
 	rc = tso_start(&state, efx, skb);
 	if (rc)
 		goto mem_err;
